include/Options/Business/Nav_Menus.php

```diff
@@ -73,6 +83,78 @@ protected function get_data_structure(): array {
 		);
 	}
 
+	/**
+	 * Sanitizes option's value.
+	 * Can populate the `$errors` property with blocking and non-blocking errors: in case of non-blocking errors,
+	 * the value is sanitized and can be stored.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array   $value   Value to sanitize.
+	 * @param Options $options All options.
+	 * @return array|WP_Error The sanitized value. An instance of `WP_Error` in case of blocking error.
+	 *
+	 * @phpstan-return NavMenusValue|WP_Error
+	 */
+	protected function sanitize( $value, Options $options ) {
+		global $polylang;
+
+		if ( empty( $polylang ) || ! $polylang->model->are_languages_ready() ) {
+			// Access to global `$polylang` is required.
+			_doing_it_wrong(
+				__METHOD__,
+				esc_html( sprintf( 'The option \'%s\' cannot be set before the hook \'pll_init\'.', static::key() ) ),
+				'3.7'
+			);
+			/** @phpstan-var NavMenusValue */
+			return $this->get();
+		}
+
+		// Sanitize new value.
+		$value = parent::sanitize( $value, $options );
+
+		if ( is_wp_error( $value ) ) {
+			// Blocking error.
+			return $value;
+		}
+
+		/** @phpstan-var NavMenusValue $value */
+		if ( empty( $value ) ) {
+			// Nothing to validate.
+			return $value;
+		}
+
+		$unknown_langs  = array();
+		$language_terms = wp_list_pluck( $this->get_language_terms(), 'slug' );
+
+		foreach ( $value as $theme_slug => $menu_ids_by_location ) {
+			foreach ( $menu_ids_by_location as $location => $menu_ids ) {
+				// Make sure the language slugs correspond to an existing language.
+				$value[ $theme_slug ][ $location ] = array();
+
+				foreach ( $language_terms as $lang_slug ) {
+					if ( ! empty( $menu_ids[ $lang_slug ] ) ) {
+						$value[ $theme_slug ][ $location ][ $lang_slug ] = $menu_ids[ $lang_slug ];
+					}
+				}
+
+				// Detect unknown languages.
+				$unknown_langs = array_merge( $unknown_langs, $menu_ids );
```

Comment:
A this point, the name of the variable `$unknown_langs` is confusing (yes I struggled to understand). We really know they are unknown only the line after.

Reply:
You're right. Improved in [6114d37](https://github.com/polylang/polylang/pull/1610/commits/6114d37fc5993ecc0d8eab1de18502eb7ab27b57).